src/llm_vm/agents/REBEL/agent.py

```diff
@@ -451,8 +451,6 @@ def promptf(self, question, memory, facts, level, split_allowed=True, max_level=
             a tuple containing the gpt response and a list with the conversation history
         """
 
-        if nlp is None:
-            nlp=spacy.load("en_core_web_md")
```

Comment:
you should use this style code up above,  nlp is supposed to be a shared variable ... what happened to it?